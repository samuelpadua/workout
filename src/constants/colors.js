export default {
  default: '#0086FF',
  success: '#00D604',
  warning: '#FF8A00',
  error: '#FF253A',
  lightGray: '#E5E5E5',
  gray: '#8C8C8C',
  mineShaft: '#404040'
}
