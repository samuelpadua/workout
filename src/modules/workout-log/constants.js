export const WORKOUT_LOG_ADD = 'workoutLog/ADD'
export const WORKOUT_LOG_REMOVE = 'workoutLog/REMOVE'
