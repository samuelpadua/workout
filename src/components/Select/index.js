import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { Flex, Box } from '@rebass/grid'
import { colors } from '../../constants'

const SelectStyled = styled.select`
  position: relative;
  border-radius: 4px;
  background: transparent;
  border: ${
  props => (props.error
    ? `2px solid ${colors.error}`
    : `1px solid ${colors.gray}`)
};
  font: inherit;
  color: currentColor;
  width: 100%;
  height: 3em;
  margin: 0;
  padding: ${
  props => (props.error
    ? '5px 11px 6px'
    : '6px 12px 7px')
};

  &:focus, &:active {
    border-color: red;
    border: 2px solid ${
  props => (props.error
    ? colors.error
    : colors.default)
};
    padding: 5px 11px 6px;
    box-shadow: none !important;
    outline: 0;
    box-shadow: none !important;
  }
`

const LabelStyled = styled.label`
  font-size: 14px;
  font-weight: 700;
  color: ${colors.mineShaft};
`

const LabelErrorStyled = styled.span`
  font-size: 14px;
  color: ${colors.error};
`

function Select(props) {
  const {
    options,
    label,
    error,
    errorMessage,
  } = props

  return (
    <Flex flexDirection="column">
      <Box mb="4px">
        <LabelStyled>
          { label }
        </LabelStyled>
      </Box>
      <Box>
        <SelectStyled {...props}>
          <option>Selecione</option>
          {
            options.map(({ value, text }) => (
              <option key={value} value={value}>{text}</option>
            ))
          }
        </SelectStyled>
      </Box>
      {error && (
        <Box mt="4px">
          <LabelErrorStyled>{ errorMessage }</LabelErrorStyled>
        </Box>
      )}
    </Flex>
  )
}

Select.propTypes = {
  options: PropTypes.arrayOf(PropTypes.shape({
    value: PropTypes.string.isRequired,
    text: PropTypes.string.isRequired,
  })).isRequired,
  label: PropTypes.string,
  error: PropTypes.bool,
  errorMessage: PropTypes.string,
}

Select.defaultProps = {
  label: '',
  error: false,
  errorMessage: '',
}

export default Select
