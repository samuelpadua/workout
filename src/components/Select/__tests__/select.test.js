
import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import Select from '../index'
import { colors } from '../../../constants'

const selectOptions = [
  {
    value: 'run',
    text: 'Run'
  },
  {
    value: 'swimming',
    text: 'Swimming'
  },
  {
    value: 'bike',
    text: 'Bike'
  }
]

describe('Select', () => {
  describe('default', () => {
    it('should render Select', () => {
      const tree = renderer
        .create(<Select options={selectOptions} />)
        .toJSON()
  
      expect(tree).toMatchSnapshot()
    })
  
    it('should change color border on focus or active', () => {
      const tree = renderer.create(<Select options={selectOptions} />).toJSON()
      expect(tree).toHaveStyleRule('border', `2px solid ${colors.default}`, {
        modifier: ':focus',
      })

      expect(tree).toHaveStyleRule('border', `2px solid ${colors.default}`, {
        modifier: ':active',
      })
    })
  })

  describe('error', () => {
    it('should render Select with error state', () => {
      const tree = renderer
        .create(<Select options={selectOptions} error />)
        .toJSON()
  
      expect(tree).toMatchSnapshot()
    })

    it('should change color border on focus or active error state', () => {
      const tree = renderer.create(<Select options={selectOptions} error />).toJSON()
      expect(tree).toHaveStyleRule('border', `2px solid ${colors.error}`, {
        modifier: ':focus',
      })

      expect(tree).toHaveStyleRule('border', `2px solid ${colors.error}`, {
        modifier: ':active',
      })
    })
  })
})
