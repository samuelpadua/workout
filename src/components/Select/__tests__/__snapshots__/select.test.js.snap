// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select default should render Select 1`] = `
.c1 {
  box-sizing: border-box;
  margin-bottom: 4px;
}

.c3 {
  box-sizing: border-box;
}

.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c4 {
  position: relative;
  border-radius: 4px;
  background: transparent;
  border: 1px solid #8C8C8C;
  font: inherit;
  color: currentColor;
  width: 100%;
  height: 3em;
  margin: 0;
  padding: 6px 12px 7px;
}

.c4:focus,
.c4:active {
  border-color: red;
  border: 2px solid #0086FF;
  padding: 5px 11px 6px;
  box-shadow: none !important;
  outline: 0;
  box-shadow: none !important;
}

.c2 {
  font-size: 13px;
  font-weight: 700;
  color: #404040;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <label
      className="c2"
    />
  </div>
  <div
    className="c3"
  >
    <select
      className="c4"
    >
      <option>
        Selecione
      </option>
      <option
        value="run"
      >
        Run
      </option>
      <option
        value="swimming"
      >
        Swimming
      </option>
      <option
        value="bike"
      >
        Bike
      </option>
    </select>
  </div>
</div>
`;

exports[`Select error should render Select with error state 1`] = `
.c1 {
  box-sizing: border-box;
  margin-bottom: 4px;
}

.c3 {
  box-sizing: border-box;
}

.c5 {
  box-sizing: border-box;
  margin-top: 4px;
}

.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c4 {
  position: relative;
  border-radius: 4px;
  background: transparent;
  border: 2px solid #FF253A;
  font: inherit;
  color: currentColor;
  width: 100%;
  height: 3em;
  margin: 0;
  padding: 5px 11px 6px;
}

.c4:focus,
.c4:active {
  border-color: red;
  border: 2px solid #FF253A;
  padding: 5px 11px 6px;
  box-shadow: none !important;
  outline: 0;
  box-shadow: none !important;
}

.c2 {
  font-size: 13px;
  font-weight: 700;
  color: #404040;
}

.c6 {
  font-size: 13px;
  color: #FF253A;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <label
      className="c2"
    />
  </div>
  <div
    className="c3"
  >
    <select
      className="c4"
    >
      <option>
        Selecione
      </option>
      <option
        value="run"
      >
        Run
      </option>
      <option
        value="swimming"
      >
        Swimming
      </option>
      <option
        value="bike"
      >
        Bike
      </option>
    </select>
  </div>
  <div
    className="c5"
  >
    <span
      className="c6"
    >
      
    </span>
  </div>
</div>
`;
